src/amo/components/Addon/index.js

```diff
@@ -65,17 +68,33 @@ export class AddonBase extends React.Component {
   }
 
   componentWillMount() {
-    const { addon, dispatch, errorHandler, params } = this.props;
+    const {
+      addon,
+      dispatch,
+      errorHandler,
+      params,
+    } = this.props;
 
     if (addon) {
       dispatch(setViewContext(addon.type));
+      dispatch(fetchOtherAddonsByAuthors({
+        addonType: addon.type,
+        authors: this.getAuthorUsernames(addon),
+        errorHandlerId: errorHandler.id,
+        slug: addon.slug,
+      }));
     } else {
       dispatch(fetchAddon({ slug: params.slug, errorHandler }));
     }
   }
 
-  componentWillReceiveProps({ addon: newAddon, params: newParams }) {
+  componentWillReceiveProps(nextProps) {
     const { addon: oldAddon, dispatch, errorHandler, params } = this.props;
+    const {
```

Comment:
I mean we use this syntax often enough but it's actually fewer lines and arguably easier to grok to read:

```js
const newAddon = nextProps.addon;
const newAddonsByAuthors = nextProps.addonsByAuthors;
const newParams = nextProps.params;
```

but just being silly really, this is fine... 😆